e2e/libs/page-manager.ts

```diff
@@ -0,0 +1,251 @@
+import { Browser, ConsoleMessage, launch, LaunchOptions, Page, Request } from 'puppeteer';
+import {
+  canLogResponse,
+  describeJsHandle,
+  getRequestData,
+  isApiFailure,
+  isWorkbenchRequest,
+  logError,
+  shouldSkipApiResponseBody,
+  transformResponseBody
+} from './page-events-helper';
+import { logger } from './logger';
+import { defaultLaunchOptions } from './page-options';
+import fs from 'fs-extra';
+import { signIn, signInWithAccessToken } from 'utils/test-utils';
+import { savePageToFile, takeScreenshot } from 'utils/save-file-utils';
+
+const failScreenshotDir = 'logs/screenshot';
+const failHtmlDir = 'logs/html';
+const { CIRCLE_BUILD_NUM } = process.env;
+const userAgent =
+  'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) ' +
+  'Chrome/80.0.3987.149 Safari/537.36' +
+  (CIRCLE_BUILD_NUM ? ` (circle-build-number/${CIRCLE_BUILD_NUM})` : '');
+
+/**
+ * Launch new Chrome.
+ * @param launchOpts: {@link LaunchOptions} New browser launch options.
+ * @return {@link Browser}
+ */
+export const launchBrowser = async (launchOpts?: LaunchOptions): Promise<Browser> => {
+  return launch(launchOpts ? launchOpts : defaultLaunchOptions);
+};
+
+/**
+ * Launch new incognito page.
+ * @param browser: {@link Browser}
+ * @return Page
+ */
+export const launchPage = async (browser: Browser): Promise<Page> => {
```

Comment:
`launchBrowser` returns an instance of Puppeteer browser and `launchPage` returns page. Situation may arise in future that needs these two func. In most cases, `withBrowser` and `withPage` func should be used.